src/api/config.ts

```diff
@@ -709,6 +711,31 @@ function spawnGit(args: string[]): Promise<void> {
   });
 }
 
+function checkoutShaOrPr(sha: string, pr: number, dir: string): Promise<void> {
+  return new Promise((resolve, reject) => {
+    let fetch = null;
+    let checkout = null;
+    let gitDir = `--git-dir ${dir}/.git`;
+
+    if (pr) {
+      fetch = `${gitDir} fetch origin pull/${pr}/head:pr${pr}`;
```

Comment:
If this commands are being used across abstruse app it would be better to create blueprints of this reusable commands. Would be more pro.